### PR TITLE
Implement new vector opcodes

### PR DESCRIPTION
**Pull request by gita-omr**
- Handle new vector opcodes introduced by OMR

Depends on https://github.com/eclipse/omr/pull/6458

### CODE REVIEW

**Comment by gita-omr** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#6458

**Comment by gita-omr** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#6458

**Comment by gita-omr** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#6458

**Comment by gita-omr** [comment]
Changed to WIP to fix some preexisting bug but related to this PR: we create Int opcodes for Byte and Short inside VectorAPIExpansion. It worked with old opcodes since type was derived from the children but not now.

@fyi: @midronij @BradleyWood @Akira1Saitoh @r30shah

**Comment by gita-omr** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#6458

**Comment by gita-omr** [comment]
Jenkins test sanity all jdk8,jdk11 depends https://github.com/eclipse/omr/pull/6458

**Comment by gita-omr** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#6458

**Comment by gita-omr** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#6458

**Comment by gita-omr** [comment]
Jenkins test sanity aarch64 jdk8,jdk11 depends eclipse/omr#6458

**Comment by gita-omr** [comment]
Jenkins test sanity aarch64_mac jdk8,jdk11 depends eclipse/omr#6458

**Comment by gita-omr** [comment]
Some testing is still going on but ready for review.

**Comment by gita-omr** [comment]
aarch64_mac failures look like the ones in https://github.com/eclipse-openj9/openj9/issues/14892

**Comment by gita-omr** [comment]
Found a problem. Fixing.

**Comment by gita-omr** [comment]
Some old but exposed now problem in VectorAPIExpansion fixed.  I think it's ready for review.

**Comment by IBMJimmyk** [comment]
I looked over the P stuff and the SPMDParallelizer changes and they seem good to me.

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#6458

**Comment by 0xdaryl** [comment]
Jenkins test sanity amac jdk11 depends eclipse/omr#6458

**Comment by knn-k** [comment]
All the failures on amac are caused by the `-Xrs` option.
https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_sanity.functional_aarch64_mac_Personal/27/tapResults/
They have been already resolved by https://github.com/eclipse/omr/pull/6467.
https://github.com/eclipse/omr/pull/6458 needs rebasing to reflect the fix.

**Review by 0xdaryl — APPROVED** [review]

**Comment by 0xdaryl** [comment]
Lest the information be lost, the acceptance build with the dependent OMR change was successful.  The only failures were the Mac M1 failures Konno-san mentioned above.

**Comment by gita-omr** [comment]
Addressed one last comment

File: runtime/compiler/optimizer/IdiomRecognition.cpp [hunk 1]
```diff
@@ -133,15 +133,15 @@ TR_CISCNode::initializeMembers(uint32_t opc, uint16_t id, int16_t dagId, uint16_
 bool
 TR_CISCNode::isEqualOpc(TR_CISCNode *t)
    {
-   //TR_ASSERT((int)TR::NumIlOps == TR_variable, "assumption for reducing compilation time");
-   static_assert((int)TR::NumIlOps == TR_variable,
+   //TR_ASSERT((int)OMR::ILOpCode::NumAllIlOps == TR_variable, "assumption for reducing compilation time");
```
[the remark this answers is not in the document]
Comment (gita-omr) [reply runtime/compiler/optimizer/IdiomRecognition.cpp]: done

**Comment by 0xdaryl** [comment]
Previous change was removing a comment.  No need to re-run CI.

**Pull request merged**